src/JMS/Serializer/GraphNavigator.php

```diff
@@ -151,7 +151,12 @@ public function accept($data, array $type = null, Context $context)
                     // metadata for the actual type of the object, not the base class.
                     if (class_exists($type['name'], false) || interface_exists($type['name'], false)) {
                         if (is_subclass_of($data, $type['name'], false)) {
-                            $type = array('name' => get_class($data), 'params' => array());
+                            if (in_array('Doctrine\Common\Proxy\Proxy', class_implements($data), true)) {
```

Comment:
the thing is that `DoctrineProxySubscriber` is an optional dependency, that acts as any other event listener, is not in the "core" of the application, and can be even put as external package.

Currently is part of the official package just for simplicity